starknet_py/net/full_node_client.py

```diff
@@ -93,6 +95,44 @@ async def get_block_traces(
     ) -> BlockTransactionTraces:
         raise NotImplementedError()
 
+    async def get_events(
+        self,
+        from_block_number: Optional[Union[int, Tag]] = None,
+        from_block_hash: Optional[Union[Hash, Tag]] = None,
+        to_block_number: Optional[Union[int, Tag]] = None,
+        to_block_hash: Optional[Union[Hash, Tag]] = None,
+        contract_address: Optional[Hash] = None,
+        keys: Optional[List[Hash]] = None,
+    ) -> Events:
+        # pylint: disable=too-many-arguments
+        params = {
+            "chunk_size": 1024,
+            "from_block": get_block_identifier(from_block_hash, from_block_number),
+            "to_block": get_block_identifier(to_block_hash, to_block_number),
+        }
+
+        if contract_address:
+            params["address"] = _to_rpc_felt(contract_address)
+        if keys:
+            params["keys"] = list(map(_to_rpc_felt, keys))
+
+        res = await self._client.call(
+            method_name="getEvents",
+            params={"filter": params},
+        )
+        ret = cast(Events, EventsSchema().load(res, unknown=EXCLUDE))
+        con_token = res.get("continuation_token")
+        while con_token:
+            params["continuation_token"] = con_token
+            res = await self._client.call(
+                method_name="getEvents",
+                params={"filter": params},
+            )
+            new_events = cast(Events, EventsSchema().load(res, unknown=EXCLUDE))
+            ret.events.extend(new_events.events)
+            con_token = res.get("continuation_token")
+        return ret
```

Comment:
One problem I can see with this approach is that it could lead to a lot of queries being done by a single `get_events` call.

We should consider adding an option to this method to not follow `continuation_token` until all possible events were consumed. But that can be addressed in the separate PR.

cc @drknzz

Reply:
#811